src/Kestrel.Core/Internal/Http/HttpHeaders.cs

```diff
@@ -45,6 +46,10 @@ public abstract class HttpHeaders : IHeaderDictionary
                 {
                     ThrowHeadersReadOnlyException();
                 }
+                if (string.IsNullOrEmpty(key))
+                {
+                    throw new InvalidOperationException(CoreStrings.InvalidEmptyHeaderName);
```

Comment:
Use a not returning `ThrowXxx` method; rather than loading string and throwing inline.